evalml/automl/automl_search.py

```diff
@@ -495,96 +517,111 @@ def search(self, data_checks="auto", show_iteration_plot=True):
-
-            # Different size indicates early stopping
-            if len(current_batch_pipeline_scores) != current_batch_size:
-                break
-
+            try:
+                new_pipeline_ids = self._engine.evaluate_batch(current_batch_pipelines)
+                loop_interrupted = False
+            except KeyboardInterrupt:
+                loop_interrupted = True
+                if self._handle_keyboard_interrupt():
+                    break
+            full_rankings = self.full_rankings
+            current_batch_idx = full_rankings['id'].isin(new_pipeline_ids)
+            current_batch_pipeline_scores = full_rankings[current_batch_idx]['score']
+            if len(current_batch_pipeline_scores) and current_batch_pipeline_scores.isna().all():
+                raise AutoMLSearchException(f"All pipelines in the current AutoML batch produced a score of np.nan on the primary objective {self.objective}.")
+
+        self.search_duration = time.time() - self._start
         elapsed_time = time_elapsed(self._start)
         desc = f"\nSearch finished after {elapsed_time}"
         desc = desc.ljust(self._MAX_NAME_LEN)
         logger.info(desc)
 
-        best_pipeline = self.rankings.iloc[0]
-        best_pipeline_name = best_pipeline["pipeline_name"]
         self._find_best_pipeline()
-        logger.info(f"Best pipeline: {best_pipeline_name}")
-        logger.info(f"Best pipeline {self.objective.name}: {best_pipeline['score']:3f}")
+        if self._best_pipeline is not None:
+            best_pipeline = self.rankings.iloc[0]
+            best_pipeline_name = best_pipeline["pipeline_name"]
+            logger.info(f"Best pipeline: {best_pipeline_name}")
+            logger.info(f"Best pipeline {self.objective.name}: {best_pipeline['score']:3f}")
         self._searched = True
 
     def _find_best_pipeline(self):
         """Finds the best pipeline in the rankings
         If self._best_pipeline already exists, check to make sure it is different from the current best pipeline before training and thresholding"""
+        if len(self.rankings) == 0:
+            return
         best_pipeline = self.rankings.iloc[0]
         if not (self._best_pipeline and self._best_pipeline == self.get_pipeline(best_pipeline['id'])):
             self._best_pipeline = self.get_pipeline(best_pipeline['id'])
             if self._train_best_pipeline:
                 X_threshold_tuning = None
                 y_threshold_tuning = None
                 X_train, y_train = self.X_train, self.y_train
-                if self.optimize_thresholds and self.objective.is_defined_for_problem_type(ProblemTypes.BINARY) and self.objective.can_optimize_threshold and is_binary(self.problem_type):
+                if is_binary(self.problem_type) and self.objective.is_defined_for_problem_type(self.problem_type) \
+                   and self.optimize_thresholds and self.objective.can_optimize_threshold:
```

Comment:
I tweaked the set of criterion we use here. This should apply for timeseries binary classification as well as plain binary classification. @freddyaboulton FYI